crates/tower-cmd/src/secrets.rs

```diff
@@ -67,61 +72,71 @@ pub fn secrets_cmd() -> Command {
-                    "Preview".bold().yellow().to_string(),
-                ],
-                secrets.iter().map(|sum| {
-                    vec![
-                        sum.name.clone(),
-                        sum.environment.clone(),
-                        sum.preview.dimmed().to_string(),
-                    ]
-                }).collect(),
-            ),
-            Err(err) => return output::tower_error(err),
-        }
+    let params = ListSecretsParams {
+        environment: Some(env.clone()),
+        all: Some(*all),
+        page: None,
+        page_size: None,
     };
 
-    output::table(headers, data);
+    match default_api::list_secrets(configuration, params).await {
+        Ok(response) => {
+            if let Some(list_response) = response.entity {
+                match list_response {
+                    default_api::ListSecretsSuccess::Status200(list_response) => {
+                        let (headers, data) = if *show {
+                            (
+                                vec![
+                                    "Secret".bold().yellow().to_string(),
+                                    "Environment".bold().yellow().to_string(),
+                                    "Value".bold().yellow().to_string(),
+                                ],
+                                list_response.secrets.iter().map(|secret| {
+                                    vec![
+                                        secret.name.clone(),
+                                        secret.environment.clone(),
+                                        secret.preview.clone(),
+                                    ]
+                                }).collect(),
+                            )
+                        } else {
+                            (
+                                vec![
+                                    "Secret".bold().yellow().to_string(),
+                                    "Environment".bold().yellow().to_string(),
+                                    "Preview".bold().yellow().to_string(),
+                                ],
+                                list_response.secrets.iter().map(|secret| {
+                                    vec![
+                                        secret.name.clone(),
+                                        secret.environment.clone(),
+                                        secret.preview.dimmed().to_string(),
+                                    ]
+                                }).collect(),
+                            )
+                        };
+                        output::table(headers, data);
+                    },
+                    default_api::ListSecretsSuccess::UnknownValue(_) => {
+                        output::failure("Received unknown response format from server");
+                    }
+                }
+            }
+        },
+        Err(err) => output::tower_error(err),
+    }
```

Comment:
There's a lot going on here, maybe it should be broken up more? Separate the outer pattern matching from the formatting perhaps?